### ISSUE TITLE
Help with understanding how to use pytestarch

### ISSUE THREAD

**Issue by felipecao** [zyskarch/pytestarch#95]
Hello, there :) 

First of all, just wanted to say thanks for all the effort put into building this project, this is much appreciated!

Reading the documentation and trying it on a real project, I'm struggling to understand how to make it work with pytest.

This is the file structure I have in place: 
```
./app
./app/jobs
./app/jobs/email_processing.py
./app/repositories
./app/repositories/db
./app/repositories/db/account_repository.py
```

Looking into the codebase, `email_processing.py` has an import pointing to `account_repository.py`: `from app.repositories.db.account_repository import AccountRepository`

Then I created a test file called `test_architectural_rules.py` to assert that dependencies in the opposite direction (repository -> jobs) should be forbidden, which looks like below:
```
import os

from pytestarch import get_evaluable_architecture, Rule

evaluable = get_evaluable_architecture('<project root>', '<project root>/app')


def test_that_repositories_dont_depend_on_blueprints():
    rule = (
        Rule()
        .modules_that()
        .are_named("app.jobs.email_processing")
        .should_not()
        .be_imported_by_modules_that()
        .are_sub_modules_of("app.repositories.db")
    )

    rule.assert_applies(evaluable)
```

Assuming I understood the documentation correctly, in theory this should work. But instead I'm getting the error below. What am I doing wrong here? :( 

This is the stacktrace I get when running the test: 
```
test_architectural_rules.py:9 (test_that_repositories_dont_depend_on_blueprints)
self = <networkx.classes.digraph.DiGraph object at 0x111b49fa0>
n = 'app.jobs.email_processing'

    def successors(self, n):
        """Returns an iterator over successor nodes of n.
    
        A successor of n is a node m such that there exists a directed
        edge from n to m.
    
        Parameters
        ----------
        n : node
           A node in the graph
    
        Raises
        ------
        NetworkXError
           If n is not in the graph.
    
        See Also
        --------
        predecessors
    
        Notes
        -----
        neighbors() and successors() are the same.
        """
        try:
>           return iter(self._succ[n])
E           KeyError: 'app.jobs.email_processing'

../../venv/lib/python3.9/site-packages/networkx/classes/digraph.py:899: KeyError

The above exception was the direct cause of the following exception:

    def test_that_repositories_dont_depend_on_blueprints():
        rule = (
            Rule()
            .modules_that()
            .are_named("app.jobs.email_processing")
            .should_not()
            .be_imported_by_modules_that()
            .are_sub_modules_of("app.services")
        )
    
>       rule.assert_applies(evaluable)

test_architectural_rules.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../venv/lib/python3.9/site-packages/pytestarch/query_language/rule.py:209: in assert_applies
    matcher.match(evaluable)
../../venv/lib/python3.9/site-packages/pytestarch/rule_assessment/rule_check/rule_matcher.py:57: in match
    rule_violations = self._find_rule_violations(evaluable)
../../venv/lib/python3.9/site-packages/pytestarch/rule_assessment/rule_check/rule_matcher.py:64: in _find_rule_violations
    explicitly_requested_dependencies = self._get_explicitly_requested_dependencies(
../../venv/lib/python3.9/site-packages/pytestarch/rule_assessment/rule_check/rule_matcher.py:130: in _get_explicitly_requested_dependencies
    return evaluable.get_dependencies(
../../venv/lib/python3.9/site-packages/pytestarch/eval_structure/evaluable_graph.py:43: in get_dependencies
    dependency = get_dependency_between_modules(
../../venv/lib/python3.9/site-packages/pytestarch/eval_structure/breadth_first_searches.py:14: in get_dependency_between_modules
    dependent_upon_nodes = get_all_submodules_of(graph, dependent_upon)
../../venv/lib/python3.9/site-packages/pytestarch/eval_structure/breadth_first_searches.py:205: in get_all_submodules_of
    children = graph.direct_successor_nodes(node)
../../venv/lib/python3.9/site-packages/pytestarch/eval_structure/networkxgraph.py:193: in direct_successor_nodes
    return sorted(self._graph.successors(node))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <networkx.classes.digraph.DiGraph object at 0x111b49fa0>
n = 'app.jobs.email_processing'

    def successors(self, n):
        """Returns an iterator over successor nodes of n.
    
        A successor of n is a node m such that there exists a directed
        edge from n to m.
    
        Parameters
        ----------
        n : node
           A node in the graph
    
        Raises
        ------
        NetworkXError
           If n is not in the graph.
    
        See Also
        --------
        predecessors
    
        Notes
        -----
        neighbors() and successors() are the same.
        """
        try:
            return iter(self._succ[n])
        except KeyError as err:
>           raise NetworkXError(f"The node {n} is not in the digraph.") from err
E           networkx.exception.NetworkXError: The node app.jobs.email_processing is not in the digraph.

../../venv/lib/python3.9/site-packages/networkx/classes/digraph.py:901: NetworkXError
```

Thanks a lot for any help!

**Comment by zyskarch** [comment]
Hi @felipecao, 

The module names are supposed to include the root module (this limitation is needed to be able to separate between internal and external modules), so for example:
If your `app` module's parent module is  '/some/path/root', all modules should start with 'root': "root.app.jobs.email_processing" and so on.

So in your case, there are two options to make it work:
1. Rename by adding "root." to your module names
2. Modify the evaluable to: `get_evaluable_architecture('<project root/app>', '<project root>/app')`

I hope this helps :) 
I will adapt the Readme to make this more obvious, thank you for bringing this up!

**Comment by felipecao** [comment]
Hi @zyskarch ! Thanks a lot for the feedback! 

Thanks to your tip, I was able to get rid of the exception saying that the node is not in the digraph. But I still have the feeling I'm not understanding something. Let me give you another example...

I have the following file structure: 
``` 
./app
./app/blueprints
./app/blueprints/account.py
./app/repositories
./app/repositories/db
./app/repositories/db/gmail_email_account_repository.py
``` 

`account.py` contains the following import statement: `from app.repositories.db.gmail_email_account_repository import GmailEmailAccountRepository`

Now I have the following architectural test, which I was expecting to **fail**, but instead, it passes:
``` 
def test_that_repositories_dont_depend_on_blueprints():
    rule = (
        Rule()
        .modules_that()
        .are_named("root.app.repositories.db.gmail_email_account_repository")
        .should_not()
        .be_imported_by_modules_that()
        .are_sub_modules_of("root.app.blueprints")
    )

    rule.assert_applies(evaluable)
``` 

Did I get it wrong? Wasn't it supposed to fail? How else should i have written this test to make sure it fails?

Thanks a lot!

**Comment by zyskarch** [comment]
Hi @felipecao,

no, you are absolutely right, this _should_ work the way you're expecting it. 
There was a bug in the name resolution for absolute imports, which I have (just) fixed in the new version 2.0.2.

Thank you for bringing this to my attention!

I've closed this ticket as done - but should you encounter any further issues, please feel free to re-open / open another ticket :)

**Comment by felipecao** [comment]
Thank you so much for the help!!
